### DISCUSSION TITLE
Better global project search experience

### DISCUSSION THREAD

**Discussion by Pokapiec** [zed-industries/zed#21458] [Show and tell]
### Check for existing issues

- [X] Completed

### Describe the feature

I have been using editors like PyCharm, IntelliJ, VS Code, Atom, Submite Text and Zed and to be honest the most enjoyable experience of searching in project i had using PyCharm/IntelliJ IDE's because they are very intuitive, just using arrows and Enter to go to desired search result in some specific file line. 

In Zed search works well but experience of using it is not ideal for me (and for couple of my work colleagues too). We have to use it as a separate tab in editor, we have to use some key binding to go to next/previous search result and to this day i don't know how to go to some specific file without using my mouse.

IntelliJ solution to search problem as popping-in modal that can be pinned seems much more convenient.

P.S. i just wanted to say that the whole editor is just amazing, i love using it and i'm writing this issue only because i would like to fully leave pycharm/vs code to use zed.

### Environment

Zed version: 0.163.3 (installed with `curl -f https://zed.dev/install.sh | sh`)
Operating system: Ubuntu 22.04.5 LTS x86_64
RAM: 32GB
Architecture: amd64

### If applicable, add mockups / screenshots to help present your vision of the feature

<img width="884" alt="image" src="https://github.com/user-attachments/assets/0888709b-1833-4b99-99da-0b597c5bf4cf">

**Comment by 5brian** [comment]
Similar: https://github.com/zed-industries/zed/issues/8279#issuecomment-2067376033

**Comment by irakliasatiani** [comment]
I really agree, this is my same concern, I really love it, but there are things that needs to be polished

**Comment by nesl247** [comment]
I'm not sure if it's out of scope for this, but find references should also use a modal (optionally for those who don't want it), just like jetbrains.
